### PR TITLE
create detail page for word list entries

### PR DESCRIPTION
**Pull request by JudithRe**
As per [US11](https://github.com/JudithRe/capstone-flashcard-dictionary/issues/17)
- when clicking on a word in the word list. A details page is shown
- added some more information to Entry Model so it would show the dictionary data
- back button redirects back to previous page
- Entry is now split, because the dictionary data does not get a details page.

(The Icon was not part of the user story and will be changed later, sorry!)

### CODE REVIEW

File: components/EntryContent/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import styled from "styled-components";
+
+function EntryContent({ entry, handleAddEntry, isEditMode, databaseMutate }) {
```
Comment (Kheirah) [inline components/EntryContent/index.js]: You're not using `isEditMode` and `databaseMutate`.
Either remove them if you don't need them or make use of them =)

File: components/StyledComponents/LinkWithoutDecoration.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const { default: Link } = require("next/link");
```
Comment (Kheirah) [inline components/StyledComponents/LinkWithoutDecoration.js]: That's a cumbersome way of importing the `Link` component 😆  Why don't you take this?
```suggestion
import Link from "next/link";
```

File: components/StyledComponents/LinkWithoutDecoration.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+const { styled } = require("styled-components");
```
Comment (Kheirah) [inline components/StyledComponents/LinkWithoutDecoration.js]: The same goes for the second import, plus it's a default, not a named.
```suggestion
import styled from "styled-components";
```

File: utils/studyFunctions.js [hunk 4]
```diff
@@ -47,10 +47,12 @@ function isDue(lastReview, stage) {
   }
 
   const reviewDate = new Date(lastReview);
+  console.log("reviewDate ", reviewDate);
   if (interval >= 0 && interval <= 120) {
     const dueDate = new Date(
       reviewDate.setDate(reviewDate.getDate() + interval)
     );
+    console.log("dueDate ", dueDate);
```
Comment (Kheirah) [inline utils/studyFunctions.js]: Remove `console.logs` =)

File: utils/deleteEntry.js [hunk 5]
```diff
@@ -7,7 +9,7 @@ export async function deleteEntry(_id, databaseMutate) {
 
   if (response.ok) {
     databaseMutate();
-    router.push("/words"); //to use the same function for deletion on detail page.
+    //router.push("/words"); //to use the same function for deletion on detail page.
```
Comment (Kheirah) [inline utils/deleteEntry.js]: Remove lines of code that you don't actively use. It's prone to confusion when code is left in the comments.

File: pages/words/[id].js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  StyledCenterAlign,
+  StyledSectionLeftAlign,
+} from "@/components/StyledComponents/StyledSection";
+import { getVisualDate } from "@/utils/helperFunctions";
+import { useRouter } from "next/router";
+
+import styled from "styled-components";
+
+export default function WordDetail({ wordList, databaseIsLoading }) {
+  // const [isEditMode, setIsEditMode] = useState(false);
```
Comment (Kheirah) [inline pages/words/[id].js]: Remove lines of code that are commented out.

File: pages/words/[id].js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  const router = useRouter();
+  const { id } = router.query;
+
+  if (databaseIsLoading) {
+    return <p>Loading...</p>;
+  }
+
+  const entryData = wordList.find(({ _id }) => _id === id);
```
Comment (Kheirah) [inline pages/words/[id].js:27]: What if I manually enter an `id` in the address bar that does not exist?

Your app currently throws an error in that case. Maybe you can safeguard your app against invalid ids by checking if the `entryData` is `undefined`? And if that's the case, return some JSX that says something like, "sorry, there's no entry".

**Review by Kheirah — CHANGES_REQUESTED** [review]
Very good!

Just minor things to be removed and one safeguard to be added that you need to look into. See below for more details.

File: pages/words/[id].js (hunk 7 above)
Comment (JudithRe) [reply pages/words/[id].js:27]: Yes good point. Thank you :)

**Review by Kheirah — APPROVED** [review]
Perfect! 🥳

**Pull request merged**